### DISCUSSION TITLE
Is there a limit on ArrayParameterType and/or SequenceContainer sizes?

### DISCUSSION THREAD

**Discussion by jbell-astrobotic** [yamcs/yamcs#899] [General]
I'm running into an interesting issue - first, here is the sample XTCE I am using:
```
<StringParameterType name="string64">
    <UnitSet />
    <StringDataEncoding>
        <SizeInBits>
            <Fixed>
                <FixedValue>512</FixedValue>
            </Fixed>
        </SizeInBits>
    </StringDataEncoding>
</StringParameterType>

<IntegerParameterType name="uint32" signed="false">
    <UnitSet />
    <IntegerDataEncoding sizeInBits="32" byteOrder="leastSignificantByteFirst" />
</IntegerParameterType>
            
<AggregateParameterType name="DIR_LIST_FileType">
    <MemberList>
        <Member typeRef="string64" name="DIR_LIST_EntryName" />
        <Member typeRef="uint32" name="DIR_LIST_EntrySize" />
        <Member typeRef="uint32" name="DIR_LIST_ModifyTime" />
        <Member typeRef="uint32" name="DIR_LIST_Mode" />
    </MemberList>
</AggregateParameterType>

<ArrayParameterType name="DIR_LIST_FileListType" arrayTypeRef="DIR_LIST_FileType">
    <DimensionList>
        <Dimension>
            <StartingIndex>
                <FixedValue>0</FixedValue>
            </StartingIndex>
            <EndingIndex>
                <!-- Adjust this value to update the number of entries -->
                <FixedValue>19</FixedValue>
            </EndingIndex>
        </Dimension>
    </DimensionList>
</ArrayParameterType>

<Parameter parameterTypeRef="string64" name="DIR_LIST_DirName" shortDescription=""/>
<Parameter parameterTypeRef="uint32" name="DIR_LIST_TotalFiles" shortDescription=""/>
<Parameter parameterTypeRef="uint32" name="DIR_LIST_PacketFiles" shortDescription=""/>
<Parameter parameterTypeRef="uint32" name="DIR_LIST_FirstFile" shortDescription=""/>
<Parameter parameterTypeRef="DIR_LIST_FileListType" name="DIR_LIST_FileList" shortDescription=""/>

<SequenceContainer name="DIR_LIST_TLM">
    <EntryList>
        <ParameterRefEntry parameterRef="DIR_LIST_DirName" />
        <ParameterRefEntry parameterRef="DIR_LIST_TotalFiles" />
        <ParameterRefEntry parameterRef="DIR_LIST_PacketFiles" />
        <ParameterRefEntry parameterRef="DIR_LIST_FirstFile" />
        <ParameterRefEntry parameterRef="DIR_LIST_FileList" />
    </EntryList>
    <BaseContainer containerRef="/SR/CfsV2TelemetryPacket">
        <RestrictionCriteria>
            <ComparisonList>
                <Comparison parameterRef="CCSDSAPID" value="0x7C" />
            </ComparisonList>
        </RestrictionCriteria>
    </BaseContainer>
</SequenceContainer>

```
The spacesystem is downlinking a packet that consists of an array of structs, so I am using an array of aggregate parameter types in XTCE. The spacesystem downlinks an array of size 20, so I've created an array with indices [0,19]. When parsing the container in Yamcs (v5.9.2) I am getting the following error:
<6>Exception processing packet : Error processing container: while extracting parameter /SR/FM/DIR_LIST_FileList at bit position 11680: Fixed size of string buffer exceeds the remaining size in bytes: 64>40

When I increase the ending index, I get the same exact error. However, if I decrease the ending index of the array type (I tested with an ending index of 5) I do not get an error. The first 5 entries are parsed correctly, but the spacesystem is downlinking 20 entries so 15 of them are just ignored in this case. I am curious as to why I am seeing this buffer exception for arrays of aggregate types only once I've exceeded some type of index threshold.

The total size of the packet is 1612 bytes including header information, and I've verified the full packet is being downlinked. However, regardless of setting the EndingIndex to 19 or 25 in Yamcs, the maximum size of the packet it will process seems to be 1,500 bytes. This number (1,500) is calculated by observing the hex dump in Yamcs Web.

**Comment by lorenzo-gomez-windhover** [comment]
@jbell-astrobotic 
Have look at this:https://docs.yamcs.org/yamcs-server-manual/links/udp-tm-data-link/

Specifically the section that says:

```
maxLength (integer)

    The maximum length of the packets received. If a larger datagram is received, the data will be truncated. Default: 1500 bytes

```

**Reply by jbell-astrobotic** [reply]
@lorenzo-gomez-windhover that was it.. thank you!

**Comment by merose** [comment]
Note that the maximum length of a CCSDS Space Packet is 65542 bytes. (65536
bytes in the user data area, represented by a packet length field of
0xFFFF==65536 - 1, plus the 6-byte CCSDS primary header) That's the value I
usually use when configuring the packetInputStream (for SLE telemetry
links) or the packet preprocessor (for UDP links, e.g.).

Mark
--
Mark Rose
***@***.***
925.922.7263
